identity-service/src/solana-client.js

```diff
@@ -191,19 +190,125 @@ async function createAndVerifyMessage (
   })
 
   let feePayerAccount = getFeePayerKeypair(false)
+  let signature = await sendAndSignTransaction(connection, transaction, feePayerAccount)
 
-  let signature = await solanaWeb3.sendAndConfirmTransaction(
-    connection,
-    transaction,
-    [feePayerAccount],
+  return signature
+}
+
+const getUnixTs = () => {
+  return new Date().getTime() / 1000;
+};
+
+// Adapted from mango send function
+// https://github.com/blockworks-foundation/mango-ui/blob/b6abfc6c13b71fc17ebbe766f50b8215fa1ec54f/src/utils/send.tsx#L785
+async function sendAndSignTransaction (connection, transaction, signers) {
+  let recentBlockHash = (await connection.getRecentBlockhash('confirmed')).blockhash
+  transaction.recentBlockhash = recentBlockHash
+
+  transaction.sign(signers)
+  let rawTransaction = transaction.serialize()
+  const startTime = getUnixTs();
+  const txid = await connection.sendRawTransaction(
+    rawTransaction,
     {
-      skipPreflight: false,
-      commitment: config.get('solanaTxCommitmentLevel'),
-      preflightCommitment: config.get('solanaTxCommitmentLevel')
+      skipPreflight: true,
+    },
+  );
+
+  // Anonymous function to retry sending until confirmation
+  (async () => {
+    while (!done && getUnixTs() - startTime < timeout) {
```

Comment:
Is this timeout defined?
If it's undefined - the second part will always be false
I also don't think done is defined in this scope.